crates/bevy_asset/src/lib.rs

```diff
@@ -1494,6 +1494,162 @@ mod tests {
+                Some(())
+            });
+
+            a_id
+        };
+
+        // handle is dropped, then immediately re-acquired
+        {
+            let handle = app
+                .world_mut()
+                .resource_mut::<Assets<CoolText>>()
+                .get_strong_handle(id)
+                .expect("should be able to get strong handle");
+            app.update();
+
+            {
+                let text = app.world().resource::<Assets<CoolText>>().get(&handle);
+                assert!(text.is_some());
+
+                assert!(server.is_loaded(handle.id()));
+            }
+        }
+
+        // handle is dropped
+        app.update();
+        assert!(
+            app.world().resource::<Assets<CoolText>>().get(id).is_none(),
+            "asset handles were dropped, so it should have been dropped last update"
+        );
+
+        assert!(!server.is_loaded(id));
+    }
+
+    #[test]
+    fn no_duplicate_assets() {
+        let dir = Dir::default();
+        dir.insert_asset_text(Path::new("dep.cool.ron"), SIMPLE_TEXT);
+
+        let (mut app, gated) = test_app(dir);
+        app.init_asset::<CoolText>()
+            .init_asset::<SubText>()
+            .init_resource::<StoredEvents>()
+            .register_asset_loader(CoolTextLoader)
+            .add_systems(Update, store_asset_events);
+
+        gated.open("dep.cool.ron");
+
+        let server = app.world().resource::<AssetServer>().clone();
+        {
+            let handle = server.load("dep.cool.ron");
+
+            let a_id = handle.id();
+            run_app_until(&mut app, |world| {
+                _ = get::<CoolText>(world, a_id)?;
+                let (load, _, _) = server.get_load_states(a_id).unwrap();
+                assert!(load.is_loaded());
+                Some(())
+            });
+
+            let handle = {
+                let mut texts = app.world_mut().resource_mut::<Assets<CoolText>>();
+                texts
+                    .get_strong_handle(handle.id())
+                    .expect("should be able to get strong handle")
+            };
+
+            app.update();
+
+            {
+                let handle2 = server.load::<CoolText>("dep.cool.ron");
+                assert!(
+                    handle.id() == handle2.id(),
+                    "loading the same asset twice should return the same handle"
+                );
+
+                match (handle, handle2) {
+                    (Handle::Strong(a), Handle::Strong(b)) => {
+                        assert!(
+                            Arc::ptr_eq(&a, &b),
+                            "loading the same asset twice should return the same handle"
+                        );
+                    }
+                    (a, b) => {
+                        panic!("expected strong handles, got {a:?} and {b:?}");
+                    }
+                }
+            }
+        };
+    }
+
+    #[test]
+    fn uuid_asset_strong_handle() {
```

Comment:
This test name confused me slightly because I didn't realize this PR actually removes the ability get a handle that can drop the UUID asset - what is absolutely a pro! Can we rename this to `get_strong_handle_for_uuid_asset`?

As an aside, we may want to rename `get_strong_handle` to `get_handle_from_id` - since weak handles no longer exist! That should be left to another PR though I think.